frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteGraphTableManager.java

```diff
@@ -130,7 +132,11 @@ private void refreshGraphTable(Path macheteFilePath, boolean isMacheteFilePresen
         LOG.info("Machete file (${macheteFilePath}) is empty");
       }
     }
-    graphTable.getModel().setRepositoryGraph(repositoryGraph);
```

Comment:
Hmmm okay... so essentially, `refreshGraphTable`, despite being marked as `@UIEffect`, has in fact been NOT invoked on the UI thread, right? Shouldn't `@UIEffect` annotation be then removed from this method (and probably transitively from its callers as well)?

Reply:
Also, where was `refreshGraphTable` called from when outside of UI thread? Was it from `new Task.Backgroundable -> run()` in `GitMacheteGraphTableManager#updateAndRefreshGraphTableInBackground`? I guess `Task.Backgroundable`s aren't executed on UI thread, right?

Reply:
Now I see that `GuiEffectChecker`, despite having a type hierarchy for **prohibiting execution of UI-effecting stuff outside of UI thread**, definitely lacks stuff for **prohibiting execution of non-UI-safe stuff on UI thread**. If there was a way to mark `com.intellij.openapi.progress.Task.Backgroundable#run` (or rather `com.intellij.openapi.progress.Progressive#run`) as non-UI-safe, the problem could not occur in the first place :/

Reply:
Actually, this is the same issue as https://github.com/typetools/checker-framework/issues/3253, but approached from a different direction: my issue says that we shouldn't allow certain methods to be executed on UI thread coz they might stall the UI; the problem here is that we shouldn't unsoundly assume that certain methods (like `Backgroundable#run` might be executed on UI thread while in fact they can't.

Reply:
I'm gonna raise that as separate, more serious issue (bug, not enhancent) against Checker

Reply:
You are right and you are right. Annotation can be removed now and actually there is on base to assume that backgroundable tasks are executed in UI thread (they are not). 

As a curiosity, I can tell that there is `org.jetbrains.annotations.CalledInAwt` annotation. It is kind of our `@UiEffect`.

Reply:
Actually, I discovered a possibly serious bug in (GuiEffect? maybe not only)Checker: Checker shouldn't have allowed us to override a `@SafeEffect` method (`run`) of `Progressive` as a `@UIEffect`... raising an issue with them now...

Reply:
https://github.com/typetools/checker-framework/issues/3287

Reply:
I fixed all problems with `@UIEffect` recognized so far on #257... and I yearn for https://github.com/typetools/checker-framework/issues/3253 even more now